lib/fluent/plugin/in_tail.rb

```diff
@@ -178,6 +182,25 @@ def configure(conf)
       # parser is already created by parser helper
       @parser = parser_create(usage: parser_config['usage'] || @parser_configs.first.usage)
       @capability = Fluent::Capability.new(:current_process)
+      # Need to create thread pool because #sleep should pause entire thread on enabling log throttling feature.
+      @thread_pool = nil
+      @enable_read_bytes_limit_thread_pool = false
+      if @read_bytes_limit_per_second > 0
+        if @max_thread_pool_size < 0
+          raise Fluent::ConfigError, "Specify positive number"
+        end
+        if !@enable_watch_timer
+          raise Fluent::ConfigError, "Need to enable watch timer when using log throttling feature"
+        end
+        if @read_bytes_limit_per_second < 8192
+          log.warn "Should specify greater equal than 8192. Use 8192 for read_bytes_limit_per_second"
+          @read_bytes_limit_per_second = 8192
+        end
+        if @max_thread_pool_size < 2
+          raise Fluent::ConfigError, "Specify 2 or more on max_thread_pool_size"
```

Comment:
Although the default value of `@max_thread_pool_size` is `1`, it should be `2` or more.
Otherwise users are always required to add `@max_thread_pool_size` config.

Reply:
After removing `max_thread_pool_size` parameter, this validation is not needed and ought to be removed.

Reply:
Removed in https://github.com/fluent/fluentd/pull/3185/commits/478b487383e1dec1af86cb0def2000b515eb6577.